src/routes/categories/containers/CategoriesPage.js

```diff
@@ -73,35 +102,55 @@ class CategoriesPage extends React.Component {
   }
 
   render () {
-    const url = this.getParentPath()
-    const hierarchy = new Hierarchy(this.props.path)
+    const category = CategoryModel.getCategoryByPath(this.props.categories, this.props.path)
+    if (!category) {
+      // todo show an error that the page has not been found, redirect to the notFound route
+    }
+    const children = this.props.categories.filter(_category => category.children.includes(_category.id))
 
-    // Pass data to hierarchy
-    const error = hierarchy.build(this.props.pages)
-    if (error) {
-      return <Error error={error} />
+    let Content
+    if (category.children.length === 0) {
```

Comment:
Extract this if else if else to a separate method. (No more evil let :P)

Reply:
done